doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -195,19 +285,548 @@ Mail chapel-developers@ for review and merge
+  across a large number of workstations.  If you have limited testing resources
+  available to you, you can request that a member of the core Chapel team help.
+
+.. _Test System: https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/bestPractices/TestSystem.rst
+
+.. _WIP PR:
+
+Work-in-progress pull requests (WIP PRs)
+++++++++++++++++++++++++++++++++++++++++
+
+This is a special kind of pull request that is not yet intended to be merged.
+Such PRs are created to take advantage of what the Github PR interface provides,
+such as public comment history and quick links between the WIP PR and other
+related issues and pull requests.  They allow the developer to get early
+feedback on a change.
+
+The status of the WIP PR should be clearly stated, including what steps need to
+be taken before the PR is ready for final review.
+
+It is generally advisable to "close" such PRs until they are ready for
+final review and testing, as their development can span a large period of time
+and would thus add clutter to the list of open PRs.
+
+It is perfectly acceptable to abandon such PRs (especially in favor of a cleaned
+up version of the code) when the git history becomes too large, so long as a
+link to the original PR is provided when the change is eventually merged.
+
+.. _reasonably sized:
+
+Pull request guidance
++++++++++++++++++++++
+
+* It is considered good practice to keep PRs (pull requests) to a reasonable
+  size. This ensures that the PR will be reviewed in a timely manner and will
+  receive a higher level of attention per line of code during review.
+
+  * When submitting a PR, the contributor should ask themselves if their
+    contribution can be separated into smaller logical chunks or independent
+    parts. Reviewers will also be pondering the same question and may request a
+    break up of the contribution into smaller PRs.
+
+  * Breaking up a PR can sometimes require a great deal of effort and
+    creativity, and may not be feasible at all, due to the intertwined nature
+    of the code.
+
+  * Ideally, the size of the PR should be proportional to the expected value to
+    the developer and user community. For example, a new module introduced as a
+    1000-line PR is acceptable, while a set of new tests introduced as a
+    1000-line PR is not.
+
+.. _Final merge message:
+
+Final merge message
++++++++++++++++++++
+
+- start with a single topic line with at most 75 characters
+- then have a blank line
+- then have a more detailed explanation including motivation for the
+  change and how it changes the previous behavior
+- use present tense (e.g. "Fix file iterator bug")
+- manually wrap long lines in the explanation to 75 or 80 characters
+
+It is good practice for the pull request title to match the topic line, and the
+detailed explanation to match the pull request description.  For contributors
+without merge permissions, this will allow the reviewer to merge changes with
+minimal additional support.
```

Comment:
This seems a little odd, structurally. Contributors without merge permissions won't be writing a merge message, so why would they be reading this section? Maybe we just need to be clearer that the reviewer will do the merge message. Not sure how to fix it best.